src/aics-image-viewer/components/TfEditor/index.tsx

```diff
@@ -143,68 +179,93 @@ const TfEditor: React.FC<TfEditorProps> = (props) => {
     ];
   };
 
-  const handlePlotPointerDown: React.PointerEventHandler<SVGSVGElement> = (event) => {
-    if (draggedPointIdxRef.current === null && event.button === 0) {
-      // this click is not on an existing point - create a new one
-      const [x, opacity] = mouseEventToControlPointValues(event);
-      const point = { x, opacity, color: lastColorRef.current };
+  const dragControlPoint = (draggedIdx: number, x: number, opacity: number): void => {
+    const newControlPoints = [...controlPointsRef.current];
+    const draggedPoint = newControlPoints[draggedIdx];
+    draggedPoint.x = x;
+    draggedPoint.opacity = opacity;
 
-      // add new control point to controlPoints
-      const index = d3.bisector<ControlPoint, ControlPoint>((a, b) => a.x - b.x).left(props.controlPoints, point);
-      setDraggedPointIdx(index);
+    // Remove control points to keep the list sorted by x value
+    const bisector = d3.bisector<ControlPoint, ControlPoint>((a, b) => a.x - b.x);
+    const idxLeft = bisector.left(newControlPoints, draggedPoint);
+    const idxRight = bisector.right(newControlPoints, draggedPoint);
 
-      const newControlPoints = [...props.controlPoints];
-      newControlPoints.splice(index, 0, point);
-      setControlPoints(newControlPoints);
+    if (idxLeft < draggedIdx) {
+      const numPointsToRemove = draggedIdx - idxLeft; // should almost always be 1
+      newControlPoints.splice(idxLeft, numPointsToRemove);
+
+      const newIdx = draggedIdx - numPointsToRemove;
+      setDraggedPointIdx(newIdx);
+      setSelectedPointIdx(newIdx);
+    } else if (idxRight > draggedIdx + 1) {
+      newControlPoints.splice(draggedIdx + 1, idxRight - draggedIdx - 1);
+    }
+
+    setControlPoints(newControlPoints);
+  };
+
+  const dragRampSlider = (handle: TfEditorRampSliderHandle, x: number): void => {
+    if (handle === TfEditorRampSliderHandle.Min) {
+      const max = rampRef.current[1];
+      setRamp([Math.min(x, max), max]);
```

Comment:
can you drag the min handle past max and vice versa?

Reply:
nope - this line ensures that each slider stops at the other slider's position rather than dragging past it.